### PR TITLE
CFY-1906 pass inputs as env variables

### PR DESCRIPTION
**Pull request by idanmo**

### CODE REVIEW

File: fabric_plugin/tasks.py [hunk 1]
```diff
@@ -170,7 +171,7 @@ def download_resource(resource_path, target_path=None):
         env['PATH'] = '{0}:$PATH'.format(remote_ctx_dir)
         env[CTX_SOCKET_URL] = proxy.socket_url
         for key, value in env.iteritems():
-            env_script.write('export {0}={1}\n'.format(key, value))
+            env_script.write("export {0}={1}\n".format(key, value))
```
Comment (dankilman) [inline fabric_plugin/tasks.py]: why change from `'` to `"`?

File: fabric_plugin/tasks.py [hunk 2]
```diff
@@ -185,6 +186,18 @@ def download_resource(resource_path, target_path=None):
             proxy.close()
 
 
+def _create_process_config(process, operation_kwargs):
+    env_vars = operation_kwargs.copy()
+    if 'ctx' in env_vars:
+        del env_vars['ctx']
+    env_vars.update(process.get('env', {}))
+    for k, v in env_vars.items():
+        if isinstance(v, dict) or isinstance(v, list) or isinstance(v, set):
```
Comment (dankilman) [inline fabric_plugin/tasks.py]: there is a shorter syntax for isinstance, you can have the same condition like this:
`if isinstance(v, (dict, list, set))`

File: fabric_plugin/tests/tests.py [hunk 3]
```diff
@@ -507,6 +510,43 @@ def test_run_script_download_resource(self):
...
+                    }
+                })
+            self.assertEqual(return_value if isinstance(value, basestring)
+                             else json.loads(return_value), value)
+        test('string-value')
+        test([1, 2, 3])
+        test({'key': 'value'})
+
+    def test_run_script_inputs_as_env_variables_process_env_override(self):
+        test_operation = self.test_run_script_inputs_as_env_variables.__name__
```
Comment (dankilman) [inline fabric_plugin/tests/tests.py]: I think having a duplicate function in script_test.sh is better than the logic applied here which deviates from all the rest of the tests.

File: fabric_plugin/tasks.py (hunk 1 above)
Comment (idanmo) [inline fabric_plugin/tasks.py]: My bad

File: fabric_plugin/tasks.py (hunk 2 above)
Comment (idanmo) [inline fabric_plugin/tasks.py]: nice!

File: fabric_plugin/tests/tests.py (hunk 3 above)
Comment (idanmo) [inline fabric_plugin/tests/tests.py]: done.

**Pull request merged**